src/Microsoft.Diagnostics.Runtime/src/DacInterface/MetaDataImport2.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Microsoft.Diagnostics.Runtime.Utilities;
+
+namespace Microsoft.Diagnostics.Runtime.DacInterface
+{
+    public sealed unsafe class MetaDataImport2 : CallableCOMWrapper
+    {
+        private static /*readonly*/ Guid IID_IMetaDataImport2 = new Guid("7DAC8207-D3AE-4c75-9B67-92801A497D44"/*"FCE5EFA0-8BBA-4f8e-A036-8F2022B08466"*/);
+
+        private CloseEnumDelegate? _closeEnum;
+        private EnumGenericParamsDelegate? _enumGenericParams;
+        private GetGenericParamPropsDelegate? _getGenericParamProps;
+
+        public MetaDataImport2(DacLibrary library, IntPtr pUnknown)
+            : base(library?.OwningLibrary, ref IID_IMetaDataImport2, pUnknown)
+        {
+        }
+
+        private ref readonly IMetaDataImport2VTable VTable => ref Unsafe.AsRef<IMetaDataImport2VTable>(_vtable);
+
+        public IEnumerable<int> EnumerateGenericParams(int token)
+        {
+            InitDelegate(ref _enumGenericParams, VTable.EnumGenericParams);
+
+            int[] tokens = ArrayPool<int>.Shared.Rent(32);
+            try
+            {
+                IntPtr handle = IntPtr.Zero;
+                int hr;
+
+                while ((hr = _enumGenericParams(Self, ref handle, token, tokens, tokens.Length, out int count)) >= 0 && count > 0)
+                    for (int i = 0; i < count; i++)
+                        yield return tokens[i];
+
+                CloseEnum(handle);
+            }
+            finally
+            {
+                ArrayPool<int>.Shared.Return(tokens);
+            }
+        }
+
+        public unsafe bool GetGenericParamProps(int token, out int index, out GenericParameterAttributes attributes, [NotNullWhen(true)] out string? name)
+        {
+            name = null;
+            InitDelegate(ref _getGenericParamProps, VTable.GetGenericParamProps);
+
+            int hr = _getGenericParamProps(
+                Self, token, out index, out attributes, out int owner, out _, null, 0, out int needed);
+
+            if (hr < 0)
+                return false;
+
+            string nameResult = new string('\0', needed - 1);
```

Comment:
I like this pattern.  I'll try to use this in the future for other SOS/Dac calls if I get around to it.